LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/DownstreamSender.cs

```diff
@@ -33,19 +33,22 @@ internal class DownstreamSender : IPacketForwarder
             this.logger = logger;
         }
 
-        public async Task SendDownstreamAsync(DownlinkMessage message)
+        public async Task<bool> SendDownstreamAsync(DownlinkMessage message)
         {
             if (message is null) throw new ArgumentNullException(nameof(message));
             if (message.StationEui == default) throw new ArgumentException($"A proper StationEui needs to be set. Received '{message.StationEui}'.");
 
             if (this.socketWriterRegistry.TryGetHandle(message.StationEui, out var webSocketWriterHandle))
             {
                 var payload = Message(message);
-                await webSocketWriterHandle.SendAsync(payload, CancellationToken.None);
+                var task = webSocketWriterHandle.SendAsync(payload, CancellationToken.None);
+                await task;
```

Comment:
if the task does not complete successfully, this line will throw, hence the next line would always be true IMHO?

Reply:
I don't really understand - if that line would always be true then what would be the point of exposing `IsCompletedSuccessfully()` in `ValueTask`?
I saw we already do something similar in a few places in the project e.g. in [`DeviceLoaderSynchronizer`](https://github.com/Azure/iotedge-lorawan-starterkit/blob/dev/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DeviceLoaderSynchronizer.cs#L171)

Reply:
The example of the DeviceLoaderSynchronizer is different. We do have a WhenAll, if one of them fails and exception is thrown and we catch it (if it is within the condition of the catch). When going through the list, they are in a unknown state, hence the check for the completion. Here we have a single task that we are awaiting and that will throw if there is an exception.

On a related note: what we do in the DLS is not safe either. There might be tasks completing after the fact we checked and that result will be discarded. It's a bit indeterministic. I believe a Task.WhenAny loop would be better there and then handle individual results as they are available and initialize or discard accordingly.

Reply:
Thanks for explaining. I opened a bug #1212 for the DeviceLoaderSynchronizer issue; since it's unrelated to this PR it makes sense to track it separately.

Reply:
After discussion and validation, we can close 1212 again. It's going to wait for all of them before raising the exception, hence it is deterministic. The only improvement we would get is quicker initialization.